core/src/main/kotlin/ch/ergon/dope/resolvable/clause/model/JoinClause.kt

```diff
@@ -32,37 +36,53 @@ sealed class SelectJoinClause : ISelectJoinClause {
         )
     }
 
+    constructor(joinType: String, bucket: Bucket, key: Field<out ValidType>, forBucket: Bucket, parentClause: ISelectFromClause) {
+        val parentDopeQuery = parentClause.toDopeQuery()
+        val bucketDopeQuery = bucket.toDopeQuery()
+        val keyDopeQuery = key.toDopeQuery()
+        val forBucketDopeQuery = forBucket.toDopeQuery()
+        dopeQuery = DopeQuery(
+            queryString = "${parentDopeQuery.queryString} $joinType ${bucketDopeQuery.queryString} " +
+                "ON KEY ${keyDopeQuery.queryString} FOR ${forBucketDopeQuery.queryString}",
+            parameters = parentDopeQuery.parameters + bucketDopeQuery.parameters + keyDopeQuery.parameters,
```

Comment:
you forgot to pass the params of the _forBucketDopeQuery_ - even though we know it is empty right now, it can change in the future

Reply:
done